File: linkedLists/reverseList.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def reverseList(self, head):
        rList = None
        while head:
            tmp = head
            head = head.next
            tmp.next = rList
            rList = tmp
        return rList

File: cpp/top100InterviewQs/linkedlist/reverse_list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head)
    {
        if (!head || !head->next) return head;
        ListNode *rhead = nullptr;
        while (head) {
            ListNode *ptr = head;
            head = head->next;
            ptr->next = rhead;
            rhead = ptr;
        }
        return rhead;
    }
};
